app/src/main/java/com/duckduckgo/app/email/waitlist/WaitlistWorkRequestBuilder.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.email.waitlist
+
+import android.content.Context
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.ListenableWorker
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequest
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.duckduckgo.app.email.AppEmailManager
+import com.duckduckgo.app.email.EmailManager
+import com.duckduckgo.app.global.plugins.worker.WorkerInjectorPlugin
+import com.duckduckgo.app.notification.NotificationSender
+import com.duckduckgo.app.notification.model.WaitlistCodeNotification
+import com.duckduckgo.di.scopes.AppObjectGraph
+import com.squareup.anvil.annotations.ContributesMultibinding
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class WaitlistWorkRequestBuilder @Inject constructor() {
+
+    fun waitlistRequestWork(withDelay: Boolean = true): OneTimeWorkRequest {
+        val requestBuilder = OneTimeWorkRequestBuilder<EmailWaitlistWorker>()
+            .setConstraints(networkAvailable())
+            .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, 1, TimeUnit.HOURS)
```

Comment:
NIT: I think the backoff criteria will never be applied since we always return success and we retry manually.

Reply:
Fair point, removed.